deepspeed/runtime/engine.py

```diff
@@ -125,21 +125,21 @@ def __init__(self,
         if dist_init_required is None:
             dist_init_required = not dist.is_initialized()
 
-        if self._in_aml():
-            self._set_environment_variables_for_nccl_backend(args)
-        else:
-            self._mpi_check(args, dist_init_required)
-
         self.dist_backend = "nccl"
+
         if dist_init_required:
```

Comment:
If `dist_init_required=False` we should ensure that `dist.is_initialized()==True` maybe via an assert?

Reply:
Got it. How about doing the following as there are only three cases, 1) None, 2) False, 3) True, rigth?

```python
        if dist_init_required is None:
            dist_init_required = not dist.is_initialized()

        if dist_init_required is False:
            assert (dist.is_initialized()==True), "Torch distributed not initialized. Please set dist_init_required to True or initialize before calling deepspeed.initialize()"

        if dist_init_required:

```